diff --git a/main.c b/main.c
index 2c3fb48..51d3a23 100644
--- a/main.c
+++ b/main.c
@@ -93,7 +93,7 @@ void handle_orientation(enum Orientation orientation) {
     // transform touch devices
     // (and pray that our lord and savior vaxry won't change hyprctl output)
     system_fmt("while IFS=$'\n' read -r device ; do "
-            "hyprctl keyword device:\"$device\":transform %d; "
+            "hyprctl keyword device:name=\"$device\":transform %d; "
             "done <<< \"$(hyprctl devices | awk '/Touch Device at|Tablet at/ {getline;print $1}')\"",
             orientation);
 }
